topo/node/srl/srl_test.go

```diff
@@ -310,3 +310,100 @@ func TestResetCfg(t *testing.T) {
 		})
 	}
 }
+
+func TestConfigPush(t *testing.T) {
+	ki := fake.NewSimpleClientset(&corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "pod1",
+		},
+	})
+
+	reaction := func(action ktest.Action) (handled bool, ret watch.Interface, err error) {
+		f := &fakeWatch{
+			e: []watch.Event{{
+				Object: &corev1.Pod{
+					Status: corev1.PodStatus{
+						Phase: corev1.PodRunning,
+					},
+				},
+			}},
+		}
+		return true, f, nil
+	}
+	ki.PrependWatchReactor("*", reaction)
+
+	ni := &node.Impl{
+		KubeClient: ki,
+		Namespace:  "test",
+		Proto: &topopb.Node{
+			Name:   "pod1",
+			Vendor: topopb.Vendor_NOKIA,
+			Config: &topopb.Config{},
+		},
+	}
+
+	tests := []struct {
+		desc     string
+		wantErr  bool
+		ni       *node.Impl
+		testFile string
+		cmdFile  string
+	}{
+		{
+			// successfully configure certificate
+			desc:     "success",
+			wantErr:  false,
+			ni:       ni,
+			testFile: "configpush_success",
+			cmdFile:  "configpush_success_cli.cfg",
+		},
+		{
+			// device returns an error during config push -- we expect to fail
+			desc:     "failure",
+			wantErr:  true,
+			ni:       ni,
+			testFile: "configpush_failure",
+			cmdFile:  "configpush_fail_cli.cfg",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			nImpl, err := New(tt.ni)
+
+			if err != nil {
+				t.Fatalf("failed creating srlinux node")
+			}
+
+			n, _ := nImpl.(*Node)
+
+			n.testOpts = []scrapliutil.Option{
+				scrapliopts.WithTransportType(scraplitransport.FileTransport),
+				scrapliopts.WithFileTransportFile(tt.testFile),
+				scrapliopts.WithTimeoutOps(2 * time.Second),
+				scrapliopts.WithTransportReadSize(1),
+				scrapliopts.WithReadDelay(0),
+				scrapliopts.WithDefaultLogger(),
+			}
+
+			if scrapliDebug() {
+				li, _ := scraplilogging.NewInstance(
```

Comment:
I'm assuming the ignored value is an error? If so it should not be ignored but instead logged using t.Log or warn or whatever to let the tester know that the scrapli logging could not be created (without needing to fail the test)

Reply:
I have added a check for errors with a test abort, as I think if the logging was required and it doesn't work, it makes sense to see what is wrong with the dependency